### PR TITLE
Allows to run tests for modules Javascript code

### PR DESCRIPTION
**Pull request by Ferdev**
Allows executing unit tests for Javascript code within modules. All javascript unit tests inside any of Jetpack modules will be run after calling the command 'yarn test-modules` in the terminal.

This PR is a suggestion from @ehg, and the code is based on the work done by @c-shultz in the PR #9144 (thanks!)

#### Changes proposed in this Pull Request:

* Added a new script entry to `package.json` to run modules Javascript unit tests.

#### Testing instructions:

* Just execute `yarn test-modules` in Jetpack base folder.
* The command shouldn't fail (command exit status should be 0).
* The output should be:

  ```  0 passing (1ms)```

  since there aren't any Javascript unit tests inside any modules yet.
* You can also add a dummy test to a folder within the modules folder and after running `yarn test-modules` again the test should be executed.

### CODE REVIEW

File: tests/runner.js [hunk 1]
```diff
@@ -1,18 +1,18 @@
...
-	path = require( 'path' ),
 	boot = require( './boot-test' );
 
 program
-	.usage( '[options] [files]' )
-	.option( '-R, --reporter <name>', 'specify the reporter to use', 'spec' )
-	.option( '-g, --grep <pattern>', 'only run tests matching <pattern>' );
+    .usage( '[options] [files]' )
+    .option( '-R, --reporter <name>', 'specify the reporter to use', 'spec' )
+    .option( '-g, --grep <pattern>', 'only run tests matching <pattern>' );
```
Comment (ehg) [inline tests/runner.js]: Separating formatting changes into different commits would make this a little easier to review.

File: docs/development-environment.md [hunk 2]
```diff
@@ -109,14 +109,15 @@ If you're not familiar with PHP Unit Testing, you can also check [this tutorial]
 
 ## Unit-testing the JS Admin Page
 
-You can run [Mocha](https://mochajs.org/) based tests for the Admin Page source code.
+You can run [Mocha](https://mochajs.org/) based tests for the Admin Page source code and the modules Javascript code.
```
Comment (ehg) [inline docs/development-environment.md]: This is worth rewording to be more general. Also, it's under the heading `Unit-testing the JS Admin Page`

File: tests/runner.js (hunk 1 above)
Comment (Ferdev) [reply tests/runner.js]: Yeah, that makes sense. I can split this commit in two and fix formatting on a different commit if you prefer.

File: docs/development-environment.md (hunk 2 above)
Comment (Ferdev) [reply docs/development-environment.md]: Good point, I'll reword this. Thanks!

File: tests/runner.js (hunk 1 above)
Comment (Ferdev) [reply tests/runner.js]: I've moved the formatting bits to a new commit.

File: docs/development-environment.md (hunk 2 above)
Comment (Ferdev) [reply docs/development-environment.md]: Done!

**Pull request merged**